File: 5423 - Acesso móvel a sistemas de informação/porp/src/Quadrado.kt
import kotlin.math.sqrt

class Quadrado {

    private var _lado: Double

    var lado: Double
        get() {
            return _lado
        }
        set(value) {
            _lado = value
        }


    var area: Double
        get() {
            println("Get area")
            return this.lado * this.lado
        }
        set(value) {
            println("set Area")
            this.lado = sqrt(value)
        }

    constructor(lado: Double){
        this._lado = lado
    }

}

File: 11027 - Desenvolvimento de aplicações móveis (plataforma iOS)/Swift/Classes_v2/Classes_v2/Quadrado.swift
//
//  Quadrado.swift
//  Classes_v2
//
//  Created by Gonçalo Feliciano on 14/06/2024.
//

import Foundation


class Quadrado{
    /*
    var logedIn:Bool{
        didSet{
            
            if logedIn{
                // load de dados
            }else{
                // "clean" da memoria
            }
        }
        
    }
     
     */
     
    var lado:Double{
        willSet{
            print("o lado vai mudar de \(self.lado) para \(newValue)")
        }
        
        didSet{
            print("o lado foi alterado para \(self.lado) de \(oldValue)")
        }
    }
    
    
    var area: Double{
        
        get{
            pow(lado, 2)
        }
        
        set{
            self.lado = sqrt(newValue)
        }
        
    }
    
    
    var perimetro: Double{
        
        get{
            4 * lado
        }
        
        set{
            self.lado = newValue / 4
        }
        
        
    }
    
    
    
    
    
    init(lado: Double) {
        self.lado = lado
    }
    
    
    
    
}
